src/content/docs/tutorials/rust/rust_tutorial.mdx

```diff
@@ -7,181 +7,211 @@ import { Tabs, TabItem } from '@astrojs/starlight/components';
+The second and subsequent runs should be faster.
 ```bash
-cd social_network_rs
-cargo init
-```
-
-Your directory will look something like this:
-```
-tutorial_data/
-├── node/
-│   ├── user.csv
-│   └── post.csv
-└── relation/
-│   ├── FOLLOWS.csv
-│   ├── POSTS.csv
-│   └── LIKES.csv
-social_network_rs/
-├── src/
-│   ├── main.rs
-│   ├── Cargo.toml
-│   └── Cargo.lock
-└── bin
-    └── query.rs
-```
-
-Replace `main.rs` with the following code snippet:
+$ cargo run --bin kuzu_social_network
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3m 18s
+     Running `target/debug/kuzu_social_network`
+Hello, world!
+
+$ cargo run --bin create-db
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.14s
+     Running `target/debug/create-db`
+Hello, world!
+```
+
+We are now ready to start the tutorial.
+
+## Create the database
+
+Replace the contents of `src/bin/create-db.rs` with the following code snippet:
 ```rs
 use kuzu::{Connection, Database, Error, SystemConfig};
 
 fn main() -> Result<(), Error> {
-    // This is where our queries will be inputted and executed.
+    // Rest of the code goes here.
 
+    Ok(())
 }
 ```
 
-We are now ready to start the tutorial.
-
-## Create an instance of the database
-In order to create the graph in Kuzu, we first need to create a new Kuzu database. We do this by using the following commands in Rust:
+Create and connect to an empty Kuzu database:
 ```rs
-// Create an empty on-disk database and connect to it
-let db = Database::new("./social_network_rs_db", SystemConfig::default())?;
+let db = Database::new("./social_network_db", SystemConfig::default())?;
 let conn = Connection::new(&db)?;
 ```
-This will create a database directory called `./social_network_rs_db`, where our data will be stored.
-Kuzu also supports `in-memory mode` in which case the lifetime of the database will be only during the execution of the Rust program.
-For more information, refer to the example in the [create your first graph](/get-started/#index) section.
+This will create a database directory `./social_network_db` where the imported csv data will be stored.
 
-## Define schema
-The first step in building any Kuzu graph is schema definition. We will define the node and relationships tables
-as per our desired schema as follows:
+Kuzu also supports an `in-memory mode` where the database is active only during the execution of the Rust program. For more information, refer to the example in the [create your first graph](/get-started/#index) section.
+
+### Define schema
+The first step in building a Kuzu graph is to define a schema. For our example dataset, we need the following node and relationships tables:
 ```rs
-conn.query("
-CREATE NODE TABLE IF NOT EXISTS User(
-    user_id INT64 PRIMARY KEY,
-    username STRING,
-    account_creation_date DATE
-)
-")?;
-conn.query("
+conn.query(
+    "
+    CREATE NODE TABLE IF NOT EXISTS User(
```

Comment:
I think this would look a little cleaner if there wasn't an empty line before the query (same for the other similar queries).
```rs
conn.query(
    "CREATE NODE TABLE IF NOT EXISTS User(
```

Reply:
Yeah. I copied whatever `rustfmt` generated, but perhaps it's worth changing it here manually.

Reply:
I think it generated that just because it won't change the contents of strings.

Reply:
Good catch, fixed.